pkg/ipamd/ipamd.go

```diff
@@ -2107,7 +2112,7 @@ func (c *IPAMContext) hasRoomForEni() bool {
 	if c.enablePodENI && c.dataStore.GetTrunkENI() == "" {
 		trunkEni = 1
 	}
-	return c.dataStore.GetENIs() < (c.maxENI - c.unmanagedENI - trunkEni)
+	return c.dataStore.GetENIs() < (c.maxENI - c.unmanagedENI - trunkEni + c.nonZeroCardENI)
```

Comment:
nit: I would rather see an `if c.numNetworkCards > 1` condition here, as then we can add a comment explaining how we do the calculation for instance with multiple network cards